@Valve, please, assign community moderators | this PR brings FAQ and checkmarks in IssueTemplates

Aluerie opened the pull request:
This PR might look overboard to somebody from Valve. But from my perspective as someone who reads majority of posted issues, the repository really needs to advertise "more upvotes (👍) the better" concept (which is advised in README of this repo) and overall "GitHub etiquette". People who come here are generally **totally unfamiliar** with those and end up misusing/underusing the repo or even being destructive. Thus we need to promote the "rules" - this is why I:
* Made FAQ and put it right into front of `README.md` so hopefully it catches eyes of a supposed first-time visitor and the word FAQ grabs their attention into at least cross-looking at it where they can understand
  - not to post duplicates
  - upvote issues with 👍 is what this repository is about
* Added markdown warning in issues templates that kindly ask them to check if the issue they want to post is a duplicate
* Added required checkmarks for users to tick before the posting
* ❗ And finally. kindly ask Valve to assign community moderatos with reasoning below. ❗
  
 ### Why? 
 
1. **Excessive amount of duplicates**
  Example: There is 46 open duplicates of #6125 and the original issue only has 22 upvotes meaning people do not understand that they are not supposed to post duplicates, that they should just upvote the original and close their duplicate. Some even think that it's beneficial to post duplicates. 
  
2. **People do not know that they are supposed to 👍the issues.**
  Example: above^ already proves it.
   
3. **Excessive amount of troll/wrong/invalid posts and comments**
  You can say that this should not be a problem since such posts just do not get upvoted, but they make it way harder to browse the repo and upvote real stuff. If a good post gets buried away from Newest page by 10 duplicates, 3 troll posts,  2 wrong posts, 5 complainers then it's not very likely for people to notice and upvote it (or for some quick dev fix). Example: During Swag Bag period there were 100s *invalid* issues/day posted which were just people crying about deliberately deleting their TB arcana thinking gem is tradable.

4. **People do not check back on their issues**
  Infinite amount of issues and devs not fully triaging the repo leads to a lot of closeable issues being left behind. People don't even react to duplicate/invalid mentions and rarely remember to close their issues after their post is addressed. Examples: already mentioned 46 duplicates of  #6125, ~40 ban appeals like #7051 or #2035 which was fixed months ago and no longer relevant. Patch 7.32e fixed arcanas not counting in turbo - but will anybody of these 30 posters close the issues on their own?

5. (Edit from Feb2024) **Trolls in comments**
  While point 3 is about troll posters - there is also a group of people sabotaging comment sections with trolling/negativity/toxicity/etc. Even non-trolling people sometimes behave like it's a reddit-like-forum but ideally the bug tracker should have more serious structure. 
  
These 5 problems are related to each other and one proposed way of trying to address it - this PR brings the FAQ into better spotlight and a bit more aggressively suggest the poster to double check their intention.

# @ Valve, please, assign some community moderators. 

 ### Why? 
 
 1. **It's just another and more efficient way of solving those mentioned 4 problems above.**
  Example: moderators would close some obvious duplicates/trolls/etc. There are 5k+ open issues and this number can be easily trimmed down. Less open issues - easier to browse, easier to search, easier to upvote. Maybe, stale settings need to be discussed, but my point is rather about Newest page. Keeping it as clean as possible would help a lot. 
2. **__Trusted__ Moderators could utilize a lot of unused GitHub features.**

---

Cmon, guys, the bitter fact is - in 7 months there was zero improvement in repository structure, let the folks do it :c So add some people who can and care as collaborators (or let s start with less permissions) to this repository. 

From current README.md:
> This tracker is only as good as you make it,

With collaboration/moderation permissions folks like us could make it a lot better. 🙃

Aluerie committed:
Adding FAQ to README.md

Aluerie committed:
Adding "Before making a post" notice to issue form templates

Aluerie committed:
Fix typos

shaileshgarg commented:
Adding to the above points
* A lot of posts are related to people not knowing how to achieve some of the baisc stuff inside the dota client, some of the ones that I have encountered are people not knowing how to unlock styles , how to unbundle a set from the inventory etc. We should look into ways for people to know how to achieve these basic things inside the client
* Many posts are issues that have been fixed, but the original Author haven't checked back on them.

Also, for issues we should look into creating TAGS like QOL, BUG-LOW, BUG-MEDIUM, BUG-HIGH sort of thing to priotise the issues.

shaileshgarg commented:
Some folks that have regularly contributed to the bug tracker, and have contributed to the issues in a positively

@TheUnholyWrath

Aluerie committed:
Adding wyk article into FAQ

commentator24 commented:
ty for tagging me. I also would love to work for free to learn until hopefully hired by valve and become valve developer / make my own custom game as well.

Would be nice if there's a guide on how to make proposed changes for entry level developers. e.g. to fix text based incorrect tooltips that has old values. if someone could guide me e.g. over discord / steam voice chat i would love to make such guides for others or learn how to make custom game modes / new maps for dota. How nice would it be if developers can just look at proposed changes first? or has guides on how to create custom games to test/fix bugs in workshop tool? and new developers can efficiently fix bugs with guides on how to fix similar type of bugs. e.g. if want to change this tooltip, need to change this and that as well, etc for files that anyone has the permission to look at.

New issues templates are not efficient / too bulky / people not using right keywords in title that would make it easier to search for similar issues later on.
![image](https://user-images.githubusercontent.com/109324155/212134119-4286259e-721f-4311-9170-3c75a7b95f1c.png)
Every time I looked at these templates i feel I wasted few seconds to look for which one to get started, so I just open blank issue instead, since clicking get started just leads to similar prompts: Description, game id, screenshots.

Templates for creating new issues should catch people's attention to upvote if that's important. noone would open readme files. e.g. the first time i created github account i don't even know how to start an upvote emote and often just create blank issue template because i feel it doesn't fit with other templates, annoyed with requirements e.g. game ID, and generally prefers the visual style of blank template preview (no bulky headings of Descriptions, example match ID and screenshots).

so my suggestions for new issues template:
1. Configure blank issue template? Add the right Alt text / input tooltip suggestions for titles so its easier to search for similar issues later on and remind people to upvote.
![image](https://user-images.githubusercontent.com/109324155/212149294-2cd34369-4168-4408-a4db-290962c33195.png)

2. Set templates that would fit assigned task / groups in Valve team / remind people to upvote, so senior developers or github moderators won't waste time assign roles / find and close similar issues / reduce the need for github moderators to change titles and its easier to search for titles with similar issues.
maybe something like this?
![image](https://user-images.githubusercontent.com/109324155/212180019-ac9cfc7f-8dcd-4018-82e0-0b987911f26d.png)

Then you can list the prompts needed for issues with similar titles to prevent duplicates.
e.g. Guide for new issue title to prevent duplicates...could be simply list Task name - where and what in titles.
No right click UI - cant gift this item (name).
Incorrect display value - tooltip show old value.. or alt text does not display correct damage etc.
this way if someone want to fix similar issue for other items or tooltips, they can just search the task name..No right click UI or incorrect display value for similar tasks.
And If something is fixed, developer should at least comment on the issue they looked at, so other similar issue can be closed as well because it's tagged under comments or by other github moderators etc.

QOL and feature suggestions should be under discussions maybe.

Aluerie committed:
Rename issue templates to rearrange their order

Aluerie commented:
> Configure blank issue template? (and some of your other talks)

Issue templates have a lot of potential. For example, one unused in this repository is auto-assigning **labels** that make searching/sorting/glance value way better. So instead of removing templates in a favor of blank issue - i would rather see improvement in current situation.

So yeah, long story short - repository organisation can be way better if we use the whole potential of GitHub.

Aluerie committed:
Numbering Typo. Oups.

shaileshgarg commented:
Rules for Community Moderators if they are ever assigned:

* **Duplicates**: Close all duplicates except the latest. Keeping latest duplicate for sorting issues by Created Date
* **Bug**: Upvote them, maybe add a tag.
* **Troll**: Close them
* **Intended Interaction**: Close them with Explanation.
* **Unknown**: Let them bake in for some time, 1 month or so, see if there are any other update, or duplicates for it. If no close it. BTW, add a tag, maybe someone else knows, or a developer directly replies.
* **QOL/Suggestion**: If its small enough, add a tag and upvote them. If big then close the issue, and direct them to Discussions.

Aluerie commented:
> maybe add a tag 
once again, ideally, there exists auto-labeling in issue templates (which currently is not set) so it would ease these "maybe add a tag" out :D  
> let them bake for some time
there is auto-stale setting as well. JeffHill was intending to play around with it, but all this time - it was not touched :c 

but well, yeah, again, same point - I guess it would be nice to have active trusted community mods to be able to discuss how to tune/manage these settings fast.

shaileshgarg commented:
Auto Labing would not work, given people are creating bugs, intented behavior, trolls and all issues under the same thing. And also bugs themselves have a differnt priority for resolution some bugs are Critical, some High, medium or low priority.


Auto stale setting is also not usefull here, as many issues open from June are actual bugs that are still present in the game and there are also trolls posts that exist from June.

Aluerie committed:
Adding checkmarks

Adding id

Blank form should not exist.

Adding checkmarks

Wrong indents. Oups.

Checked twice, but always something is wrong on the first commit. :c

Uniting templates

Removing bug auto-label since it's the only template

Update bug.yaml

Update bug.yaml

Revert "Update bug.yaml"

This reverts commit 41d90d1b8d2fed0640f9002b2b8b1fb19920dc6c.

Dropdown > Required

Aluerie commented:
Added required checkmarks to issue templates.

PS. Maybe I'm moving in wrong direction with this. I'm just editing it as I like. 

Maybe, I need some confirmation/opinions on what I'm doing, so please, check my fork https://github.com/Aluerie/Dota2-Gameplay for suggestions/corrections :x

shaileshgarg commented:
A template I think should be present for QOL or Small Suggestion.

The templates with Auto Labelling works, for the General bug template, we would need to evaluate over time, what catagories of bugs are created using it and maybe create new template as and when required. Overwise the templates with Autolabelling looks good.

Some Exampes of issues which would not fit in the templates defined and fall in General Bug:
* Hero Shard Issues
* Battlepass issue
* Replay/ Dota TV issues
* Dota Client Issues, etc

Don't know how many geneuin issues are there of the above mentioned categories, but like I said above we would need to evaluate these over time.


Still not in favour of Auto Stale, as with Upvotes, there are many issues open from June which are still present in the game but with a low number of upvotes.

Aluerie committed:
New templates

shaileshgarg reviewed (commented):
Can we also add a template for small QOL improvements.

Aluerie committed:
Adding QoL template

Aluerie commented:
* Made some new templates (and QoL suggestions too) according to your suggestions 
* Removed Ability-Draft (I don't think I saw any AD bug reports - and people can just Morph/Rubick it out into normal scenario anyway) 

---
But now New Issue page feels a bit cluttered.  I'm not really sure what to do about it. Possible solution is just to have one template "Bug" and then there would be dropdown question to choose category bug. 

Pros: 
  * we only have one template

Cons: 
  * That would leave the answer to be left in the body (like checklist won't go away itself)
  * auto-labelling will have to be from GitAction that scans the body which is not a reliable way.

shaileshgarg commented:
We should freeze any more changes and finalize the pull request. 
The main purpose of the changes were:
* Have better auto labelling, the changes acheives that
* Prevent people from trolling/ using it as support forum, which the changes reflect.
* Preventing people from creating duplicates and upvoting the original, this is not entirely solved as people may create the bug with different wording but with Community mods this can be solved.


One last ask, in new issues, it is possible to start the issue with one upvote already present, because it seems people don't know how to add one.

Aluerie committed:
Addng note to upvote self

Aluerie commented:
> Have better auto labelling, the changes acheives that

Contributors of this repository still have to add the labels matching names in issue templates manually for auto-labeling to work.

In fact auto-labeling was implemented 7 months ago, labels matching those were never added by contributors.

przemek-git commented:
As much as I appreciate efforts to make this issue tracker a better place, I must say all this might be just a waste of time. Issue tracker is an experiment, and it was already reviewed how it's going. At this point it might be already abandoned, there might be enough issues to solve anyway, there might be not enough resources to work on posted issues. We know nothing and even though it is said:
> This tracker is only as good as you make it, and so far it's been a very valuable tool for making Dota a better game.

it makes no sense to make it better **until someone from Valve decides** it's worth the effort.

All in all, what's the point of having long list of issues, if nobody is ever going to pick it up? If you see any sense in having one, fell free to open your own bug tracker, and spam pull requests/discussions/issues here to advertise it - can't you see that this repository is already flooded by bots advertising different sites? A lot of comments might look like they make sense, but it's just hidden ad posted by bot.

My comment might sound pessimistic, but I want to point out again, that this repo is an experiment, and as mentioned above it is/was a very valuable tool. I had the same idea of helping community a bit by commenting under people's issues - how to post them properly or marking duplicates, but I've dropped it after one comment for reasons above. 

If Valve makes a call to moderate this PR or use any other community-driven issue tracker, then count me in, but until this happens, I suggest everyone to spent their time on something else, prolly doing overwatch sessions in dota2 or creating content to promote game will help more.

Aluerie committed:
Adding link to Linux/Mac specific bugtracker

PeruSonOfBitches commented:
Peru whores have no conflicts with apes, coz they are exactly the same. If you don't block Peru players from accessing the US servers. You are son of peru whores.

PeruSonOfWhores commented:
Para hablar como un peruano, puedes usar algunas de estas expresiones y modismos que son comunes en el español peruano:

"Bueno, por más que agradezco los esfuerzos por mejorar este rastreador de problemas, debo decir que todo esto podría ser solo una pérdida de tiempo. El rastreador de problemas es un experimento, y ya se revisó cómo va. En este punto, podría estar abandonado, podría haber suficientes problemas para resolver de todas maneras, podría no haber suficientes recursos para trabajar en los problemas publicados. No sabemos nada y, aunque se dice que:

no tiene sentido mejorarlo hasta que alguien de Valve decida que vale la pena el esfuerzo.

En resumen, ¿cuál es el punto de tener una larga lista de problemas si nadie va a ocuparse de ellos? Si ves algún sentido en tener uno, siéntete libre de abrir tu propio rastreador de errores y publicar solicitudes de extracción / discusiones / problemas aquí para publicitarlo, ¿no ves que este repositorio ya está inundado de bots que anuncian diferentes sitios? Muchos comentarios pueden parecer que tienen sentido, pero en realidad son anuncios ocultos publicados por bots.

Mi comentario puede sonar pesimista, pero quiero señalar nuevamente que este repositorio es un experimento, y como se mencionó anteriormente, es / fue una herramienta muy valiosa. Tenía la misma idea de ayudar a la comunidad un poco comentando en los problemas de las personas: cómo publicarlos correctamente o marcar duplicados, pero lo abandoné después de un comentario por las razones mencionadas anteriormente.

Si Valve decide moderar este repositorio o utilizar cualquier otro rastreador de problemas impulsado por la comunidad, entonces cuénten conmigo, pero hasta que eso suceda, sugiero que todos dediquen su tiempo a algo más, probablemente hacer sesiones de overwatch en Dota2 o crear contenido para promocionar el juego será más útil.

soy una puta peruana, y mi madre también

I am a Peru whore, and so is my mom.

ivs123franco commented:
Issue template and also feedbacks, community mods and easier access to the community such as discord would tackle the entry barrier for new players to post here. 

I did this post: https://github.com/ValveSoftware/Dota2-Gameplay/issues/6650 , 37 people watched I think... But I never had a feedback about that, I have no idea if I did it the right way. 

I think so many people want to improve the community, but they dosnt know how to or if they do... They just have no feedback and quit on trying to. 

We should increase the feedback rate and one way to do that is to bring "Overwatch from dota 2" to here, with rewards such as medals in github in order to be recognized by the community after reviewing a forum or providing a good idea.

Aluerie committed:
Note about search, upvote and faq link

shaileshgarg commented:
Would be great if there is any resolution on this. Currently there are more than 6500 issues are open, where majority are duplicates, fixed issues, complaints, no more applicable issues.

Aluerie commented:
https://github.com/ValveSoftware/Dota2-Gameplay/issues/created_by/Under-Mod

Prime example why this repository needs community moderation. Valve, please...

![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/33165440/1d72de25-3e45-431e-9c82-915eeb6b2aab)

@kisak-valve I'm sorry to @ you like this... I know you don't particularly moderate current repository, but 

* the person in screenshot is egregiously spamming and disrupting. Can something be done? 😞 
* Can this PR be referenced to somebody from Valve because this repository really needs some love and moderation from community itself.

Anutrix commented:
Can we please ban https://github.com/ValveSoftware/Dota2-Gameplay/issues/created_by/Under-Mod and remove all the duplicates?

Philaeux commented:
I am working on a new tool to manage issues related to Dota, should be in open beta soon. 

My motivation comes from the fact that some people want to improve the game they are playing by curating known issues. Right now, this repository feels like a throw away were players can toss bugs, but nothing is curated. Developers, if they check the issues, will lose a lot of time finding real bugs. It's precious time lost not fixing the bugs.

I'm trying to merge features similar to github issues (create issues, categories, filtering, comments) with some social features (rate issue impact, propose issue patches, vote on those  patches) so that the community can curate the Bug Repository, and the developer can quickly focus on what's important.

Will report when it's live with the URL.

Philaeux commented:
Informing you that I was able to build a first version of the tool I was explaining before, hosted at [oBugs](https://obugs.the-cluster.org/s/DOTA2).  
I plan on increasing the features and scale the VM to match the charge. Hoping we can curate the bug repository with this kind of tools.

BledNavalny commented:
Devs dont care about dota players, only issues they can solve its cosmetic bugs. They just hide under steam support, who kick off all here to some mystical wise devs who allegedly watch these threads

Blu3cr0ss commented:
Hello from 2024 huh... This PR shows how devs interested in fixing bugs. Still beta (most likely forever lol)

CREEATION commented:
Hello everyone,

in advance, I was very pleased to [read](https://github.com/ValveSoftware/Dota2-Gameplay/assets/3277769/9c8f765b-39df-4950-a984-943aedf89b31) [the latest patch notes](https://www.dota2.com/patches/7.36b).

I remembered I made a [reddit post](https://www.reddit.com/r/DotA2/comments/fg87x6/bug_clockwerks_power_cogs_dont_cancel_snapfires/) about how odd I thought it was that Clockwerk's Power Cogs don't cancel Snapfire's Mortimer Kisses. Not even quite sure it's related to the latest patch as I haven't played a lot in recent years plus a lot of things changed.

Issue was, I didn't remember when or with which account I posted it. So, I used the search function and stumbled upon [this reddit post](https://www.reddit.com/r/DotA2/comments/10nim6h/guys_github_bug_tracker_is_not_reddit_stop/) with a PSA that the github bug tracker is not reddit. The rabbit hole thickens and here I am.

Instinctively, I clicked on the pull requests and found this one.

Maybe the issue is that not enough people know about this repository to report actual bugs, which could help with managing the repository by having more accurate and less non-issues in the first place. Could backfire for sure, but maybe it's worth a shot to actually link the issues page from within the Dota 2 Client.

Or, even better, have a dedicated gameplay feedback window somewhere instead of the kinda hidden "[Send Feedback](https://github.com/ValveSoftware/Dota2-Gameplay/assets/3277769/e0a54e48-9657-4c76-a7de-c0b61f8ce7d3)" link which opens a simple [text box popup](https://github.com/ValveSoftware/Dota2-Gameplay/assets/3277769/0321ab99-7061-4b22-964c-e3c42443b79e).

This new gameplay feedback popup could be setup to work in tandem with the defined issue templates and automatically create an issue, maybe requiring a GitHub login. Haven't really looked into the details for doing this as I classify this comment as brainstorming.

Anyways, just a thought of thoughts. Thanks for reading! 🐢

tolik518 commented:
Duplicates are an huge issue over here, its hard to find the correct issues because there just too many. 

The up- and downvote system from github discussions and more moderator would probably already fix the current state.

LuisitoFCantagojr commented:
Nothing was fixed here!

https://github.com/ValveSoftware/Dota2-Gameplay/issues/31576
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31510
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31638
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31644
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31643
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31640
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31639
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31641
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31642
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31637
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31512
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31512
https://github.com/ValveSoftware/Dota2-Gameplay/issues/31713
